modules/raw-rsa-keyring-node/test/raw_rsa_keyring_node.test.ts

```diff
@@ -0,0 +1,182 @@
+
+describe('RawRsaKeyringWebCrypto::constructor', () => {
+  const keyName = 'keyName'
+  const keyNamespace = 'keyNamespace'
+
+  it('constructor decorates', async () => {
+    const test = new RawRsaKeyringNode({
+      rsaKey: {
+        privateKey: privatePem,
+        publicKey: publicPem
+      },
+      keyName,
+      keyNamespace
+    })
+
+    expect(test.keyName).to.equal(keyName)
+    expect(test.keyNamespace).to.equal(keyNamespace)
+    expect(test._wrapKey).to.be.a('function')
+    expect(test._unwrapKey).to.be.a('function')
+    expect(test).to.be.instanceOf(KeyringNode)
+  })
+
+  it('can construct with only public key', () => {
+    const testPublicOnly = new RawRsaKeyringNode({
+      rsaKey: {
+        publicKey: publicPem
+      },
+      keyName,
+      keyNamespace
+    })
+    expect(testPublicOnly).to.be.instanceOf(RawRsaKeyringNode)
+  })
+
+  it('can construct with only private key', () => {
+    const testPrivateOnly = new RawRsaKeyringNode({
+      rsaKey: {
+        privateKey: privatePem
+      },
+      keyName,
+      keyNamespace
+    })
+    expect(testPrivateOnly).to.be.instanceOf(RawRsaKeyringNode)
+  })
+
+  it('Precondition: RsaKeyringNode needs either a public or a private key to operate.', () => {
+    expect(() => new RawRsaKeyringNode({
+      keyName,
+      keyNamespace,
+      rsaKey: {}
+    })).to.throw()
+  })
+
+  it('Precondition: RsaKeyringNode needs identifying information for encrypt and decrypt.', () => {
+    // @ts-ignore Typescript is trying to save us.
+    expect(() => new RawRsaKeyringNode({
+      rsaKey: { privateKey: privatePem, publicKey: publicPem }
+    })).to.throw()
+    // @ts-ignore Typescript is trying to save us.
+    expect(() => new RawRsaKeyringNode({
+      rsaKey: { privateKey: privatePem, publicKey: publicPem },
+      keyNamespace
+    })).to.throw()
+    // @ts-ignore Typescript is trying to save us.
+    expect(() => new RawRsaKeyringNode({
+      rsaKey: { privateKey: privatePem, publicKey: publicPem },
+      keyName
+    })).to.throw()
+  })
+})
+
+describe('RawRsaKeyringWebCrypto encrypt/decrypt', () => {
+  const keyNamespace = 'keyNamespace'
+  const keyName = 'keyName'
+  const keyring = new RawRsaKeyringNode({
+    rsaKey: { privateKey: privatePem, publicKey: publicPem },
+    keyName,
+    keyNamespace
+  })
+  let encryptedDataKey: EncryptedDataKey
+
+  it('can encrypt and create unencrypted data key', async () => {
+    const suite = new NodeAlgorithmSuite(AlgorithmSuiteIdentifier.ALG_AES256_GCM_IV12_TAG16_HKDF_SHA256)
+    const material = new NodeEncryptionMaterial(suite)
+    const test = await keyring.onEncrypt(material)
+    expect(test.hasValidKey()).to.equal(true)
```

Comment:
Does this compare the type that is returned to be true? Or will any true value returned from this function pass this test?

Reply:
https://www.chaijs.com/api/bdd/#method_equal

it is `===` . The issue is that the udk should be random, so there is not too much to test.  Given that I pull the udk below maybe this has no value.
Want me to pull it out?

Reply:
No i was trying to understand JS a little better.